File: foody-master/Back End/src/main/java/com/food/service/RestaurantService.java
package com.food.service;

import java.util.List;

import com.food.model.Restaurant;

public interface RestaurantService {
	public Restaurant addRestaurant(Restaurant restaurant);
	public List<Restaurant> getAllRestaurants();
	public Restaurant getRestaurantById(int id);
	public Restaurant updateRestaurant(Restaurant restaurant);
	public void removeRestaurant(int id);
	public List<Restaurant> sortByName();
	public List<Restaurant> sortByCity();
	public List<Restaurant> sortByState();
	public List<Restaurant> sortByNameDsc();
	public List<Restaurant> sortByCityDsc();
	public List<Restaurant> sortByStateDsc();
}

File: foody-master/Front End/src/app/services/restaurant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { restaurantUrl } from '../config/api'
import { Restaurant } from '../model/restaurant';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class RestaurantService {

  constructor(private http: HttpClient) { }

  getAllRestaurants(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl);
  }

  getRestaurantById(id: number):Observable<Restaurant>{
    return this.http.get<Restaurant>(restaurantUrl+"/"+id);
  }

  sortByName(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl+"/sort/name")
  }

  sortByCity(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl+"/sort/city")
  }

  sortByState(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl+"/sort/state")
  }

  sortByNameDsc(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl+"/sort/name/dsc")
  }

  sortByCityDsc(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl+"/sort/city/dsc")
  }

  sortByStateDsc(): Observable<Restaurant[]>{
    return this.http.get<Restaurant[]>(restaurantUrl+"/sort/state/dsc")
  }
}
